src/Cake.Core.Tests/Unit/IO/PathTests.cs

```diff
@@ -101,6 +101,18 @@ public void Should_Throw_If_Path_Contains_Illegal_Characters()
                 Assert.Equal($"Illegal characters in directory path (*).{Environment.NewLine}Parameter name: path", result?.Message);
             }
 
+            [WindowsFact]
+            public void Should_Throw_If_Path_Contains_Illegal_Characters_Non_GlobberSymbol()
+            {
+                // Given
+                var result = Record.Exception(() => new TestingPath("hello/A|B/world.txt"));
+
+                // Then
+                Assert.IsType<ArgumentException>(result);
+                Assert.Equal("path", ((ArgumentException)result)?.ParamName);
+                Assert.Equal($"Illegal characters in directory path (|).{Environment.NewLine}Parameter name: path", result?.Message);
```

Comment:
Should be `Illegal characters in path (|)` since it might be a file path (`FilePath`).